### PR TITLE
54 factoryinput

### LINKED ISSUE #54: Create backend for UI to change data (factoryinput)
To achieve the overarching epic we need a backend / REST API that sends data to MQTT. It will be consumed by the Grafana datasource and is called factoryinput. It will be packaged together with factoryinsight.

The openapi specification can be found here:

https://github.com/united-manufacturing-hub/united-manufacturing-hub/blob/3ea6310b0477933f4495da01fbf1f6c3d1402bde/docs/static/openapi/factoryinput.yml

### PR DESCRIPTION
**Pull request by Scarjit**
Fixes #54
Also fixes related bugs in grafana-proxy

### CODE REVIEW

**Comment by Scarjit** [comment]
> SonarCloud Quality Gate failed.    Quality Gate failed
> 
> [Bug](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=BUG) [A](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=BUG) [0 Bugs](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=BUG)
> [Vulnerability](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=VULNERABILITY) [A](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=VULNERABILITY) [0 Vulnerabilities](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=VULNERABILITY)
> [Security Hotspot](https://sonarcloud.io/project/security_hotspots?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=SECURITY_HOTSPOT) [A](https://sonarcloud.io/project/security_hotspots?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=SECURITY_HOTSPOT) [0 Security Hotspots](https://sonarcloud.io/project/security_hotspots?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=SECURITY_HOTSPOT)
> [Code Smell](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=CODE_SMELL) [A](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=CODE_SMELL) [1 Code Smell](https://sonarcloud.io/project/issues?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&resolved=false&types=CODE_SMELL)
> 
> [No Coverage information](https://sonarcloud.io/component_measures?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489) No Coverage information
> [](https://sonarcloud.io/component_measures?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&metric=new_duplicated_lines_density&view=list) [15.7% Duplication](https://sonarcloud.io/component_measures?id=united-manufacturing-hub_united-manufacturing-hub&pullRequest=489&metric=new_duplicated_lines_density&view=list)

We might want to have an UMH-Queue gomodule, and other modules to prevent re-writing tons of duplicate code

File: deployment/factorycube-server/values.yaml [hunk 1]
```diff
@@ -64,6 +72,7 @@ grafanaproxy:
   enabled: true  
   image:
     repository: unitedmanufacturinghub/grafana-proxy
+    tag: latest
```
Comment (JeremyTheocharis) [inline deployment/factorycube-server/values.yaml:75]: Please remove

File: golang/cmd/factoryinput/queue.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"encoding/json"
+	"github.com/beeker1121/goque"
+	"go.uber.org/zap"
+)
+
+const queuePath = "/data/factoryinsight/queue"
```
Comment (JeremyTheocharis) [inline golang/cmd/factoryinput/queue.go:9]: really not important, but maybe change to factoryinput

File: golang/cmd/grafana-proxy/grafana/api/user/user.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ type User struct {
 	AvatarURL      string        `json:"avatarUrl"`
 }
 
-const GrafanaUrl = "http://localhost:3000"
+const GrafanaUrl = "http://factorycube-server-grafana:8080"
```
Comment (JeremyTheocharis) [inline golang/cmd/grafana-proxy/grafana/api/user/user.go:39]: let's read in via environment variable

File: golang/cmd/grafana-proxy/http.go [hunk 4]
```diff
@@ -216,37 +236,46 @@ func HandleFactoryInput(c *gin.Context, request getProxyRequestPath, method stri
...
+
 	// Grafana sessionCookie not present in request
 	sessionCookie, err := c.Cookie("grafana_session")
 	if err != nil {
+		zap.S().Warnf("No grafana_session in cookie !")
 		c.AbortWithStatus(http.StatusUnauthorized)
 		return
 	}
 
+	zap.S().Warnf("CheckUserLoggedIn")
```
Comment (JeremyTheocharis) [inline golang/cmd/grafana-proxy/http.go:249]: remove warnings

File: golang/cmd/grafana-proxy/main.go [hunk 5]
```diff
@@ -17,7 +17,7 @@ var shutdownEnabled bool
 func main() {
 
 	// Setup logger and set as global
-	logger, _ := zap.NewProduction()
+	logger, _ := zap.NewDevelopment()
```
Comment (JeremyTheocharis) [inline golang/cmd/grafana-proxy/main.go:20]: Change back to Production

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]
Additionally, please add docs

**Pull request merged**